make_version.py

```diff
@@ -7,7 +7,15 @@ def main():
     if len(output) == 3:
         version, build, commit = output
     else:
-        raise Exception("Could not git describe, (got %s)" % output)
+        # If the clone is shallow, describe output won't have tag and
+        # number of commits.  This is a particular issue on Travis-CI,
+        # which by default clones with a depth of 50.
+        # This behaviour isn't well documented in git-describe docs,
+        # but see, e.g., https://stackoverflow.com/a/36389573/1008142
+        # and https://github.com/travis-ci/travis-ci/issues/3412
+        version = 'unknown'
+        build = 'unknown'
+        commit = output
```

Comment:
Yes, thanks -- I see control/_version.py has the line ```__commit__ = "['0ede87e']"``` in my test shallow clone.

On the ```'-'.join(output)``` form.  I don't know if describe ever puts out a line with a single dash (I couldn't find definitive documentation), but if a tag has one or more dashes, ```.split``` will return a list with *more* than 3 elements (I just tested this with a fake tag to confirm it).  I could replace the ```.split``` with ```.rsplit('-',2)``` to handle this case, and use ```'-'.join``` in the event that ```git describe``` outputs a line with only one dash.

I had never considered a security aspect here.  I would have thought that since we're handling 'git describe' output as a string, and storing its results as a string in _make_version.py, there's not an obvious way to get code injection.  (I don't know how ```__conda_version__.txt``` and ```__conda_buildnum__.txt``` are processed.)  If we postulate a compromised git, wouldn't it be possible to do code injection when ```git clone``` is executed anyway?

A final point: how useful is this elaborate versioning?  Why not just use ```'0.8.0a0+git'``` in ```setup.py``` until the 0.8.0 release is made?  Or, alternatively, ```0.8.0a0+git<commit-hash>```, which is somewhat complex, but simpler than we have now.